### PR TITLE
fcitx: Add test (Unicode input, table input, m17n)

### PR DESCRIPTION
**Pull request by asdf8dfafjk**
###### Motivation for this change
Added test for fcitx

###### Things done

 - [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
  
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/tests/fcitx/default.nix [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+            machine.sleep(1)
+
+            ### Turn off Fcitx
+            machine.send_key("ctrl-spc")
+            machine.sleep(1)
+
+            ### Redirect typed characters to a file
+            machine.send_chars(" > fcitx_test.out\n")
+            machine.sleep(1)
+            machine.screenshot("terminal_chars")
```
Comment (asdf8dfafjk) [inline nixos/tests/fcitx/default.nix:146]: Screenshot for the situation that

* THere's a failure &&
* I'm not available && 
* Someone unfamiliar with Fcitx wants to see the last good state

|| 

* Someone just wants to get a sense of what's happening

File: nixos/tests/fcitx/default.nix [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+            fcitx_confdir = "${userHome}/.config/fcitx";
+        in 
+        ''
+            # We need config files before login session
+            # So copy first thing
+
+            # Point and click would be expensive,
+            # So configure using files
+            machine.copy_from_host(
+                "${fcitx_profile_file_host}",
```
Comment (jtojnar) [inline nixos/tests/fcitx/default.nix:80]: Does

```suggestion
                "${./profile}",
```

really not work?

File: nixos/tests/fcitx/config [hunk 3]
```diff
@@ -0,0 +1,147 @@
+[Hotkey]
+# Trigger Input Method
+#TriggerKey=CTRL_SPACE
```
Comment (jtojnar) [inline nixos/tests/fcitx/config:3]: Do we need all these commented out settings?

File: nixos/tests/fcitx/default.nix (hunk 2 above)
Comment (asdf8dfafjk) [reply nixos/tests/fcitx/default.nix]: I wasn't using "${..}" but instead string name. "${./profile}" works. I don't know about `tmpfiles.d` so using the copying approach.

File: nixos/tests/fcitx/config (hunk 3 above)
Comment (asdf8dfafjk) [reply nixos/tests/fcitx/config]: I cleared them up now.

**Comment by asdf8dfafjk** [comment]
Hi, Thank you!

I made another commit, you can look at the diff of `.nix` file and look at the config, profile files in their own right (not diffs). I will squash the two commits before you merge.

**Comment by asdf8dfafjk** [comment]
ToDo: C-S-f in zhengma

**Comment by roberth** [comment]
This is great. Thank you so much @asdf8dfafjk !

**Pull request merged**